src/Grids/regular_latitude_longitude_grid.jl

```diff
@@ -0,0 +1,111 @@
+const R_Earth = 6371.0e3    # Mean radius of the Earth [m] https://en.wikipedia.org/wiki/Earth
+
+struct RegularLatitudeLongitudeGrid{FT, TX, TY, TZ, A} <: AbstractHorizontallyCurvilinearGrid{FT, TX, TY, TZ}
+        Nx :: Int
+        Ny :: Int
+        Nz :: Int
+        Hx :: Int
+        Hy :: Int
+        Hz :: Int
+        Lx :: FT
+        Ly :: FT
+        Lz :: FT
+        Δλ :: FT
+        Δϕ :: FT
+        Δz :: FT
+      λᶠᵃᵃ :: A
+      λᶜᵃᵃ :: A
+      ϕᵃᶠᵃ :: A
+      ϕᵃᶜᵃ :: A
+      zᵃᵃᶠ :: A
+      zᵃᵃᶜ :: A
+    radius :: FT
+end
+
+function RegularLatitudeLongitudeGrid(FT=Float64; size, latitude, longitude, z, radius=R_Earth, halo=(1, 1, 1))
+    @assert length(latitude) == 2
+    @assert length(longitude) == 2
+    @assert length(z) == 2
+
+    λ₁, λ₂ = longitude
+    @assert -180 <= λ₁ < λ₂ <= 180
+
+    ϕ₁, ϕ₂ = latitude
+    @assert -90 <= ϕ₁ < ϕ₂ <= 90
+
+    z₁, z₂ = z
+    @assert z₁ < z₂
+
+    TX = λ₁ == -180 && λ₂ == 180 ? Periodic : Bounded
+    TY = Bounded
+    TZ = Bounded
+    topo = (TX, TY, TZ)
+
+    Nx, Ny, Nz = N = validate_size(TX, TY, TZ, size)
+    Hx, Hy, Hz = H = validate_halo(TX, TY, TZ, halo)
+
+    Lλ = λ₂ - λ₁
+    Lϕ = ϕ₂ - ϕ₁
+    Lz = z₂ - z₁
+
+    Δϕ = Lλ / Nx
+    Δλ = Lϕ / Ny
+    Δz = Lz / Nz
+
+    # Now including halos
+    λF₋ = λ₁ - Hx * Δλ
+    ϕF₋ = ϕ₁ - Hy * Δϕ
+    zF₋ = z₁ - Hz * Δz
+
+    λF₊ = λ₂ + Hx * Δλ
+    ϕF₊ = ϕ₂ + Hy * Δϕ
+    zF₊ = z₂ + Hz * Δz
+
+    λC₋ = λF₋ + Δλ / 2
+    ϕC₋ = ϕF₋ + Δϕ / 2
+    zC₋ = zF₋ + Δz / 2
+
+    λC₊ = λF₋ + total_extent(TX, Hx, Δλ, Lλ)
+    ϕC₊ = ϕF₋ + total_extent(TY, Hy, Δϕ, Lϕ)
+    zC₊ = zF₋ + total_extent(TZ, Hz, Δz, Lz)
+
+    TFλ, TFϕ, TFz = total_length.(Face, topo, N, H)
+    TCλ, TCϕ, TCz = total_length.(Center, topo, N, H)
+
+    # FIXME? MITgcm generates (xG, yG) vorticity grid points then interpolates them to generate (xC, yC):
```

Comment:
We don't seem to be using the notation `(xG, yG)`.  Does this translate to `(xF, yF)` here?

Reply:
If `xG, yG` are the locations of vertical vorticity, then it corresponds to `xᶠᶠᶜ` in our notation.